idem_provider_azurerm/states/azurerm/keyvault/key.py

```diff
@@ -0,0 +1,314 @@
+
+    :param vault_url: The URL of the vault that the client will access.
+
+    :param key_ops: A list of permitted key operations. Possible values include: 'decrypt', 'encrypt', 'sign',
+        'unwrap_key', 'verify', 'wrap_key'.
+
+    :param enabled: Whether the key is enabled for use.
+
+    :param expires_on: When the key will expire, in UTC. This parameter must be a string representation of a Datetime
+        object in ISO-8601 format.
+
+    :param not_before: The time before which the key can not be used, in UTC. This parameter must be a string
+        representation of a Datetime object in ISO-8601 format.
+
+    :param tags: A dictionary of strings can be passed as tag metadata to the key.
+
+    :param connection_auth: A dict with subscription and authentication parameters to be used in connecting to the
+        Azure Resource Manager API.
+
+    Example usage:
+
+    .. code-block:: yaml
+
+        Ensure key exists:
+            azurerm.keyvault.key.present:
+                - name: my_key
+                - key_type: my_type
+                - vault_url: my_vault
+                - tags:
+                    contact_name: Elmer Fudd Gantry
+                - connection_auth: {{ profile }}
+
+    '''
+    ret = {
+        'name': name,
+        'result': False,
+        'comment': '',
+        'changes': {}
+    }
+
+    if not isinstance(connection_auth, dict):
+        ret['comment'] = 'Connection information must be specified via connection_auth dictionary!'
+        return ret
+
+    key = await hub.exec.azurerm.keyvault.key.get_key(
+        name=name,
+        vault_url=vault_url,
+        azurearm_log_level='info',
+        **connection_auth
+    )
+
+    if key_type != 'oct':
+        key_type = key_type.upper().replace('_', '-')
+
+    if 'error' not in key:
+        if tags:
+            tag_changes = await hub.exec.utils.dictdiffer.deep_diff(key.get('properties').get('tags', {}), tags or {})
+            if tag_changes:
+                ret['changes']['tags'] = tag_changes
+
+        if key_ops is not None:
+            if sorted(key_ops) != sorted(key.get('key_operations')):
+                ret['changes']['key_operations'] = {
+                    'old': key.get('key_operations'),
+                    'new': key_ops
+                }
+
+        if enabled is not None:
+            if enabled != key.get('properties').get('enabled'):
+                ret['changes']['enabled'] = {
+                    'old': key.get('properties').get('enabled'),
+                    'new': enabled
+                }
+
+        if expires_on:
+            if expires_on != key.get('properties').get('expires_on'):
+                ret['changes']['expires_on'] = {
+                    'old': key.get('properties').get('expires_on'),
+                    'new': expires_on
+                }
+
+        if not_before:
+            if not_before != key.get('properties').get('not_before'):
```

Comment:
`get('properties')` needs to default to an empty dict or else you get:
`TypeError: 'NoneType' object is not iterable`